provider/uprops/src/casemapping.rs

```diff
@@ -0,0 +1,203 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::uprops_serde;
+use icu_casemapping::provider::{CaseMappingV1, CaseMappingV1Marker};
+use icu_casemapping::CaseMapping;
+use icu_codepointtrie::{CodePointTrieHeader, TrieType};
+use icu_provider::prelude::*;
+
+use std::convert::TryFrom;
+use std::fs;
+use std::path::PathBuf;
+
+pub struct CaseMappingDataProvider {
+    case_mapping: CaseMapping<'static>,
+}
+
+/// A data provider reading from .toml files produced by the ICU4C icuwriteuprops tool.
+impl CaseMappingDataProvider {
+    pub fn try_new(path: PathBuf) -> Result<Self, DataError> {
+        let toml_str = fs::read_to_string(&path).unwrap();
+        let toml: uprops_serde::case::Main = toml::from_str(&toml_str).unwrap();
+
+        let trie_data = &toml.ucase.code_point_trie;
+        let trie_type: TrieType =
+            TrieType::try_from(trie_data.trie_type_enum_val).map_err(DataError::new_resc_error)?;
+        let trie_header = CodePointTrieHeader {
+            high_start: trie_data.high_start,
+            shifted12_high_start: trie_data.shifted12_high_start,
+            index3_null_offset: trie_data.index3_null_offset,
+            data_null_offset: trie_data.data_null_offset,
+            null_value: trie_data.null_value,
+            trie_type,
+        };
```

Comment:
Suggestion: It would be nice to make a utility function to convert from the TOML CPT Header to the ICU4X CPT Header.  The utility function could live in this crate for now (perhaps in a "helpers" module) and be shared between both this code and the uprops code.  I see that you already re-use `super::enumerated::EnumeratedPropertyCodePointTrie` in your serde definition, so this should be fairly easy to do.

Reply:
I went ahead and implemented TryFrom<&SerializedCodePointTrie> for CodePointTrieHeader.